diff --git a/src/rdma.c b/src/rdma.c
index 8a51331..0ac55b0 100644
--- a/src/rdma.c
+++ b/src/rdma.c
@@ -13,11 +13,21 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
+#ifndef __wasm__
 #define VALKEYMODULE_CORE_MODULE
+#endif
 #include "server.h" // Include server.h to use serverLog.
 #include "serverassert.h"
 #include "connection.h"
 
+#ifdef __wasm__
+
+int RegisterConnectionTypeRdma(void) {
+    return C_ERR;
+}
+
+#else
+
 #if defined __linux__ && defined USE_RDMA /* currently RDMA is only supported on Linux */
 #if (USE_RDMA == 1 /* BUILD_YES */) || \
     ((USE_RDMA == 2 /* BUILD_MODULE */) && defined(BUILD_RDMA_MODULE) && (BUILD_RDMA_MODULE == 2))
@@ -1935,3 +1945,5 @@ int RegisterConnectionTypeRdma(void) {
 }
 
 #endif /* __linux__ */
+
+#endif /* __wasm__ */
